File: src/leetcode/java/RelativeRanks_506.java
package leetcode.java;

import java.util.Map;
import java.util.TreeMap;

public class RelativeRanks_506 {
  public String[] findRelativeRanks(int[] nums) {
    String[] ranks = new String[nums.length];
    Map<Integer, Integer> m = new TreeMap<>();
    for (int i = 0; i < nums.length; i++) m.put(nums[i], i);
    int i = nums.length;
    for (Map.Entry<Integer, Integer> entry : m.entrySet()) {
      if (i == 3) ranks[entry.getValue()] = "Bronze Medal";
      else if (i == 2) ranks[entry.getValue()] = "Silver Medal";
      else if (i == 1) ranks[entry.getValue()] = "Gold Medal";
      else ranks[entry.getValue()] = String.valueOf(i);
      i--;
    }
    return ranks;
  }
}

File: src/leetcode/python/RelativeRanks_506.py
class RelativeRanks_506(object):
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        m = {}
        for i, n in enumerate(nums):
            m[n] = i
        sorted_keys = [m[k] for k in sorted(m.keys(), reverse=True)]
        strs = [] * len(nums)
        rank = 1
        for i in sorted_keys:
            if rank == 1:
                strs[i] = "Gold Medal"
            elif rank == 2:
                strs[i] = "Silver Medal"
            elif rank == 3:
                strs[i] = "Bronze Medal"
            else:
                strs[i] = str(rank)
            rank += 1
        return strs
